src/main/java/use_cases/add_edit_category_use_case/CategoryUCI.java

```diff
@@ -0,0 +1,153 @@
+     * NOTE: There are actually two different EntityException thrown.
+     * One is from the SessionStorage Object when checking if monthID is in session.
+     *  (Although we know MonthlyStorage with monthID is always in the SessionStorage, it will be caught at views/add_edit_category_views/AddCategoryV.java).
+     * The second one is from the creation of a Category Object implementing addCategory() from MonthlyStorage, and is caught in the current implementation!
+     *
+     * @param categoryIDAdd CategoryID required for adding a new category to the designated monthID MonthlyStorage Object.
+     * @return CategoryOD String message indicating success/fail add attempt.
+     * @throws EntityException thrown when categoryIDAdd has invalid category information.
+     */
+    @Override
+    public CategoryOD addCategoryInMonth(CategoryID categoryIDAdd)throws EntityException{
+        MonthlyStorage month = categoryIDAdd.getSession().getMonthlyData(categoryIDAdd.getMonthID());
+        try{
+            double valueDouble = toDouble(categoryIDAdd.getValue());
+            Category newCategory = new Category(categoryIDAdd.getName(), valueDouble);
+
+            if (newCategory.getBudget() < 0) {
+                // 2. Category budget less than 0 fail
+                CategoryOD categoryODFailAdd = new CategoryOD("Category budget can't be less than $0. Please try again!");
+                return categoryOB.fail(categoryODFailAdd);
+            }
+            month.addCategory(newCategory);
+
+            CategoryOD categoryODSuccessAdd = new CategoryOD("You have added a new category!");
+            return categoryOB.success(categoryODSuccessAdd);
+
+        } catch(NumberFormatException|NullPointerException e){
+            //1. NumberFormatException/NullPointerException fail
+            CategoryOD categoryODFailAdd = new CategoryOD("Category budget is needs to be a number. Please try again!");
+            return categoryOB.fail(categoryODFailAdd);
+
+        } catch (EntityException e) {
+            //3. EntityException fail
+            CategoryOD categoryODFailAdd = new CategoryOD("There is already a category with this new name in this month.");
+            return categoryOB.fail(categoryODFailAdd);
+        }
+    }
+    /**
+     * Overrides method in CategoryIB.
+     * Attempts to edit a category with information from CategoryID and returns a CategoryOD indicating whether fail/success after execution.
+     * Provides detailed fail messages according to each condition listed below:
+     *  1. Repeated Name: User tries to edit category name to another name that exists in the month.
+     *  2. NoSuchElementException: User tries to edit a category that does not exist.
+     *  3. NumberFormatException: User tries to edit a budget value with input that can not be converted to a double.
+     *  4. Category budget less than 0: User tries to edit a budget value with input that is a negative number.
+     *
```

Comment:
Same here, make the Javadocs more readable by moving them to where you've wrote the code.